File: C#/Nth Prime/NthPrime.cs
using System;

public static class NthPrime
{
    public static bool IsPrime(int nth)
    {
        for (int i = 2; i <= Math.Sqrt(nth); i++)
        {
            if (nth % i == 0)
            {
                return false;
            }
        }

        return nth > 1;
    }

    public static int Prime(int nth)
    {
        if(nth == 0) { throw new ArgumentOutOfRangeException(); }

        int primeCounter = 0;
        int counter = 1;
        int answer = 0;

        while (primeCounter != nth)
        {
            if (IsPrime(++counter))
            {
                answer = counter;
                primeCounter++;
            }
        }

        return answer;
    }
}

File: TypeScript/nth-prime.ts
import { isPrime } from "./helper/primes";

export const nth = (n: number): number => {
    if(n === 0) {
        throw new Error("Prime is not possible");
    }

    let primeCounter = 0;
    let counter = 1;
    let answer = 0;
    
    while(primeCounter != n) {        
        if(isPrime(++counter)) {
            answer = counter;
            primeCounter++;
        }            
    }

    return answer;
};
  
const n = 10001;

console.log(`n=${n} -> prime=${nth(n)}`);